python/ray/_raylet.pyx

```diff
@@ -1772,17 +1772,18 @@ cdef class CoreWorker:
             check_status(CCoreWorkerProcess.GetCoreWorker().KillActor(
                   c_actor_id, True, no_restart))
 
-    def cancel_task(self, ObjectRef object_ref, c_bool force_kill,
+    def cancel_task(self, object_refs, c_bool force_kill,
                     c_bool recursive):
         cdef:
-            CObjectID c_object_id = object_ref.native()
-            CRayStatus status = CRayStatus.OK()
-
-        status = CCoreWorkerProcess.GetCoreWorker().CancelTask(
-                                            c_object_id, force_kill, recursive)
-
-        if not status.ok():
-            raise TypeError(status.message().decode())
+            CObjectID c_object_id
+            CRayStatus status
+
+        for object_ref in object_refs:
+            c_object_id = (<ObjectRef>object_ref).native()
+            status = CCoreWorkerProcess.GetCoreWorker().CancelTask(
+                        c_object_id, force_kill, recursive)
+            if not status.ok():
+                raise TypeError(status.message().decode())
```

Comment:
I'm not sure if short-circuit is a good semantics here.

Several semantics I can think of:
1. current PR behavior
2. all or nothing: seems current cancel is best-effort and all errors are raised before sending the cancel request so we should be able to achieve all or nothing semantics.
3. try to cancel all, raise an error if any fails.

My preference is (3 or 2) > 1

Also a side note: we should improve the doc for ray.cancel to make it clear about the semantics.

Reply:
I made updates in this part. Now `CancelTask` is called on every `object_ref` and the returned status is pushed into a vector. Once done, we visit the `statuses` vector and raise error if any `status` is not `ok`.

Reply:
Other approach is to accumulate all the failure messages in a single string and the raise a `TypeError` with that single accumulated string as the error message. What do you think?